api/domain_handlers.go

```diff
@@ -381,3 +382,94 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 
 	writeJSONOrBust(w, issues)
 }
+
+// RemoveIneligibleDomains runs eligibility checks on domains present in the
+// database and change the status to PendingAutomatedRemoval if the domain
+// does not follow the requirements for more than 2 crawls
+
+// Example: GET /removeineligibledomains?
+func (api API) RemoveIneligibleDomains(w http.ResponseWriter, r *http.Request) {
+
+	// domain states of domains with valid policyTypes
+	var policyStates []database.DomainState
+	// names of all the domains with valid policyTypes
+	var policyDomains []string
+	// all domains that need to be added to the ineligible
+	// domain database
+	var ineligibleDomains []database.IneligibleDomainState
+	// all domains that need to be deleted from the
+	// ineligible domain database
+	var deleteEligibleDomains []string
+
+	// Get all domains
+	domains, err := api.database.AllDomainStates()
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not retrieve domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+
+	// Filter Domains
+	// #TODO: parallelize this for loop
+	for _, d := range domains {
+		if d.Policy == preloadlist.Bulk18Weeks || d.Policy == preloadlist.Bulk1Year {
+			policyDomains = append(policyDomains, d.Name)
+			policyStates = append(policyStates, d)
+		}
+	}
+
+	// call GetIneligibleDomainStates, add to map
+	var states map[string]database.IneligibleDomainState
+	state, err := api.database.GetIneligibleDomainStates(policyDomains)
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not get domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+	for _, s := range state {
+		states[s.Name] = s
+	}
+
+	// Store ineligible domains in slice
+	for _, d := range policyStates {
+		_, issues := api.hstspreload.EligibleDomain(d.Name, d.Policy)
+
+		scan := database.Scan{
+			ScanTime: time.Now(),
+			Issues:   issues,
+		}
+
+		val, ok := states[d.Name]
+		if len(issues.Errors) > 0 {
+			if ok {
+				val.Scans = append(val.Scans, scan)
+				ineligibleDomains = append(ineligibleDomains, val)
+			} else {
+				ineligibleDomains = append(ineligibleDomains, database.IneligibleDomainState{
+					Name:   d.Name,
+					Scans:  []database.Scan{scan},
+					Policy: string(d.Policy),
+				})
+			}
+		} else if ok {
+			deleteEligibleDomains = append(deleteEligibleDomains, d.Name)
+		}
+	}
+
+	// Delete eligible domains from the database
+	err = api.database.DeleteIneligibleDomainStates(deleteEligibleDomains)
```

Comment:
For domains that are in `states` but aren't in `policyStates`/`policyDomains`, we should remove them from the ineligible domain state table at some point. (This could happen if a domain stops meeting the preload requirements, gets scanned and that result is marked down, and then the domain owner fully removes the domain from the preload list.)